httpbin/core.py

```diff
@@ -409,35 +410,22 @@ def hidden_basic_auth(user='user', passwd='passwd'):
 @app.route('/digest-auth/<qop>/<user>/<passwd>')
 def digest_auth(qop=None, user='user', passwd='passwd'):
     """Prompts the user for authorization using HTTP Digest auth"""
-    if qop not in ('auth', 'auth-int'):
-        qop = None
-    if 'Authorization' not in request.headers or  \
+
+    if 'Authorization' not in request.headers or \
                        not check_digest_auth(user, passwd) or \
                        not 'Cookie' in request.headers:
-        response = app.make_response('')
-        response.status_code = 401
-
-        # RFC2616 Section4.2: HTTP headers are ASCII.  That means
-        # request.remote_addr was originally ASCII, so I should be able to
-        # encode it back to ascii.  Also, RFC2617 says about nonces: "The
-        # contents of the nonce are implementation dependent"
-        nonce = H(b''.join([
-            getattr(request,'remote_addr',u'').encode('ascii'),
-            b':',
-            str(time.time()).encode('ascii'),
-            b':',
-            os.urandom(10)
-        ]))
-        opaque = H(os.urandom(10))
-
-        auth = WWWAuthenticate("digest")
-        auth.set_digest('me@kennethreitz.com', nonce, opaque=opaque,
-                        qop=('auth', 'auth-int') if qop is None else (qop, ))
-        response.headers['WWW-Authenticate'] = auth.to_header()
-        response.headers['Set-Cookie'] = 'fake=fake_value'
-        return response
+        return digest_challenge_response(qop, 401)
```

Comment:
That's right. Yet, I don't see one contribution blocking another. The one
that is good to merge can be accepted and the other can adjust itself to
it. This adjustment must happen from one of the two parts anyway. Be it by
the last finished. Otherwise, if the first never finishes, the second would
never be accepted although it's already finished.
Em sáb, 7 de fev de 2015 às 13:36, Ian Cordasco <notifications@github.com>
escreveu:

> In httpbin/core.py
> <https://github.com/Runscope/httpbin/pull/206#discussion_r24290516>:
>
> > -        nonce = H(b''.join([
> > -            getattr(request,'remote_addr',u'').encode('ascii'),
> > -            b':',
> > -            str(time.time()).encode('ascii'),
> > -            b':',
> > -            os.urandom(10)
> > -        ]))
> > -        opaque = H(os.urandom(10))
> > -
> > -        auth = WWWAuthenticate("digest")
> > -        auth.set_digest('me@kennethreitz.com', nonce, opaque=opaque,
> > -                        qop=('auth', 'auth-int') if qop is None else (qop, ))
> > -        response.headers['WWW-Authenticate'] = auth.to_header()
> > -        response.headers['Set-Cookie'] = 'fake=fake_value'
> > -        return response
> > +        return digest_challenge_response(qop, 401)
>
> @reinert <https://github.com/reinert> I'd rather you not re-implement
> something that's already been implemented by someone else and is
> functionally their contribution. Please don't do that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Runscope/httpbin/pull/206/files#r24290516>.
>